docs/vcpkg-schema-definitions.schema.json

```diff
@@ -0,0 +1,715 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema",
+  "$id": "https://raw.githubusercontent.com/microsoft/vcpkg-tool/master/docs/vcpkg-schema-definitions.schema.json",
+  "definitions": {
+    "artifact-references": {
+      "description": "A key/value pair of artifact name to selected artifact version.",
+      "type": "object",
+      "minProperties": 1,
+      "patternProperties": {
+        "": {
+          "type": "string"
+        }
+      }
+    },
+    "artifact-registry": {
+      "description": "An artifact registry backed by a git repository.",
+      "type": "object",
+      "properties": {
+        "kind": {
+          "type": "string",
+          "pattern": "artifact"
+        },
+        "name": {
+          "$ref": "#/definitions/identifier"
+        },
+        "location": {
+          "type": "string",
+          "format": "uri",
+          "description": "A URI to the location of the artifact registry; typically a git URI."
+        }
+      },
+      "required": [
+        "kind",
+        "name",
+        "location"
+      ]
+    },
+    "builtin-registry": {
+      "type": "object",
+      "properties": {
+        "baseline": {
+          "$ref": "#/definitions/sha1",
+          "description": "The SHA that will be checked out to find default versions for ports from this registry."
+        },
+        "kind": {
+          "type": "string",
+          "pattern": "builtin"
+        }
+      },
+      "required": [
+        "baseline",
+        "kind"
+      ]
+    },
+    "contact": {
+      "type": "object",
+      "properties": {
+        "email": {
+          "type": "string"
+        },
+        "role": {
+          "$ref": "#/definitions/string-or-strings"
+        }
+      },
+      "required": [
+        "role"
+      ],
+      "additionalProperties": false
+    },
+    "contacts": {
+      "type": "object",
+      "patternProperties": {
+        "": {
+          "$ref": "#/definitions/contact"
+        }
+      }
+    },
+    "default-registry": {
+      "type": "object",
+      "allOf": [
+        {
+          "oneOf": [
+            {
+              "$ref": "#/definitions/builtin-registry"
+            },
+            {
+              "$ref": "#/definitions/git-registry"
+            },
+            {
+              "$ref": "#/definitions/filesystem-registry"
+            }
+          ]
+        },
+        {
+          "not": {
+            "$ref": "#/definitions/packages-list"
+          }
+        }
+      ]
+    },
+    "demands": {
+      "description": "A set of filters to conditionally apply artifact pieces. Typically contains other 'requires' or 'see-also' or similar elements. The keys are platform expressions with identifiers assumed to refer to the host.",
+      "type": "object",
+      "patternProperties": {
+        "": {
```

Comment:
We should decide whether we want to support `$....`-style comments in this object

Reply:
I said 'no' for now since I don't think it currently works, but we could fix it in the future.